### PR TITLE
[PATCH v3] example: classifier: improvements for performance testing

### PR DESCRIPTION
**Pull request by JereLeppanen**
Some improvements to the classifier example, mainly to enable performance testing.

v2:
- Rebase.
- Tuomas' comments.

v3:
- Rebase.
- Add review tags.

### CODE REVIEW

File: example/classifier/odp_classifier.c [hunk 1]
```diff
@@ -1310,7 +1310,7 @@ static void usage(void)
 		"  -c, --count <num>        CPU count, 0=all available, default=1\n"
 		"\n"
 		"  -m, --mode <mode>        0: Packet Drop mode. Received packets will be dropped\n"
-		"                           !0: Packet ICMP mode. Received packets will be sent back\n"
+		"                           !0: Reply mode. Received packets will be sent back\n"
```
Comment (TuomasTaipale) [inline example/classifier/odp_classifier.c:1339]: Nitpick: reply or *echo*?

**Review by TuomasTaipale — COMMENTED** [review]
Perhaps a more stringent CLI input validation could be considered if this is now becoming a proper tester (e.g. pool count/element count/element size capa checks, negative burst size etc.).

**Comment by JereLeppanen** [comment]
> Perhaps a more stringent CLI input validation could be considered if this is now becoming a proper tester

No more proper than it already was. I don't think we should add a bunch of input validation code unless we really need it for some reason.

File: example/classifier/odp_classifier.c [hunk 2]
```diff
@@ -1152,7 +1151,6 @@ static int parse_args(int argc, char *argv[], appl_args_t *appl_args)
 				ret = -1;
 				break;
 			}
-			policy = 1;
```
[the remark this answers is not in the document]
Comment (JereLeppanen) [reply example/classifier/odp_classifier.c:1155]: v2.

File: example/classifier/odp_classifier.c [hunk 3]
```diff
@@ -291,6 +293,10 @@ static odp_pktio_t create_pktio(const char *dev, odp_pool_t pool)
 			odp_pktio_to_u64(pktio), dev,
 			odp_pktio_to_u64(pktio));
 
+	odp_pktio_config_init(&cfg);
+	cfg.parser.layer = appl_args_gbl->parse_layer;
+	odp_pktio_config(pktio, &cfg);
```
[the remark this answers is not in the document]
Comment (JereLeppanen) [reply example/classifier/odp_classifier.c]: v2.

File: example/classifier/odp_classifier.c (hunk 1 above)
Comment (JereLeppanen) [reply example/classifier/odp_classifier.c]: Echo is probably slightly better. v2.

File: example/classifier/odp_classifier.c [hunk 4]
```diff
@@ -277,11 +281,29 @@ static odp_pktio_t create_pktio(const char *dev, odp_pool_t pool)
...
+	if (num_tx > (int)capa.max_output_queues) {
+		printf("Sharing %i output queues between %i workers\n",
+		       capa.max_output_queues, num_tx);
+		num_tx = capa.max_output_queues;
+	}
+
+	appl_args_gbl->num_pktout = num_tx;
+
+	odp_pktout_queue_param_init(&pktout_queue_param);
+	pktout_queue_param.num_queues = num_tx;
```
[the remark this answers is not in the document]
Comment (JereLeppanen) [reply example/classifier/odp_classifier.c:308]: As an option maybe. But I'm not testing that at this time, so I'm leaving it as is.

**Comment by TuomasTaipale** [comment]
About the CLI error checking, for users, it is better to have sane error messages logged by the tester logic instead of something weird or undefined behavior from implementations. Doesn't have to be this PR though.

**Review by TuomasTaipale — APPROVED** [review]

**Pull request merged**